packages/abi-utils/lib/arbitrary.ts

```diff
@@ -0,0 +1,315 @@
+              }
+        )
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}));
+
+export const Abi = () =>
+  fc
+    .tuple(
+      ConstructorEntry(),
+      FallbackEntry(),
+      ReceiveEntry(),
+      fc.array(fc.oneof(FunctionEntry(), EventEntry()))
+    )
+    .chain(([constructor, fallback, receive, entries]) =>
+      fc.shuffledSubarray([constructor, fallback, receive, ...entries])
+    );
+
+namespace Numerics {
+  // 0 < n <= 32
+  export const Bytes = () => fc.nat(31).map(k => 32 - k);
+
+  // o < n <= 256
+  export const Bits = () => Bytes().map(k => 8 * k);
+
+  // 0 < n < 80
+  export const DecimalPlaces = () => fc.nat(79).map(k => k + 1);
+
+  export const Precision = () => fc.tuple(Bits(), DecimalPlaces());
+}
+
+namespace Primitives {
+  export const Uint = () => Numerics.Bits().map(m => `uint${m}`);
+  export const Int = () => Numerics.Bits().map(m => `int${m}`);
+  export const Address = () => fc.constant("address");
+  export const Bool = () => fc.constant("bool");
+  export const Fixed = () =>
+    Numerics.Precision().map(([m, n]) => `fixed${m}x${n}`);
+  export const Ufixed = () =>
+    Numerics.Precision().map(([m, n]) => `ufixed${m}x${n}`);
+  export const BytesM = () => Numerics.Bytes().map(m => `bytes${m}`);
+  export const Function = () => fc.constant("function");
+  export const Bytes = () => fc.constant("bytes");
+  export const String = () => fc.constant("string");
+  export const Tuple = () => fc.constant("tuple");
+}
+
+const Primitive = () =>
+  fc.oneof(
+    Primitives.Uint(),
+    Primitives.Int(),
+    Primitives.Address(),
+    Primitives.Bool(),
+    Primitives.Fixed(),
+    Primitives.Ufixed(),
+    Primitives.BytesM(),
+    // Primitives.Function(), // TODO re-add
+    Primitives.Bytes(),
+    Primitives.String(),
+    Primitives.Tuple()
+  );
+
+const Type: fc.Memo<string> = fc.memo(n =>
+  n === 0
+    ? Primitive()
+    : fc.oneof(Primitive(), ArrayFixed(n > 3 ? 3 : n), ArrayDynamic(n))
+);
+
+const ArrayFixed = fc.memo(n => {
+  const tuple =
+    n <= 1
+      ? fc.tuple(Primitive(), fc.integer(1, 256))
+      : fc.tuple(Type(), fc.integer(1, 256));
+
+  return tuple.map(([type, length]) => `${type}[${length}]`);
+});
+
+const ArrayDynamic = fc.memo(n => {
+  const item = n <= 1 ? Primitive() : Type();
+
+  return item.map(type => `${type}[]`);
+});
+
+const reservedWords = new Set([
+  "after",
+  "alias",
+  "apply",
+  "auto",
+  "case",
+  "copyof",
+  "default",
+  "define",
+  "final",
+  "immutable",
+  "implements",
+  "in",
+  "inline",
+  "interface",
+  "let",
+  "macro",
+  "match",
+  "mutable",
+  "null",
+  "of",
+  "partial",
+  "promise",
+  "reference",
+  "relocatable",
+  "sealed",
+  "sizeof",
+  "static",
+  "supports",
+  "switch",
+  "typedef",
+  "typeof",
+  "unchecked"
+]);
+
+const fakerToArb = (template: string, transform = camelCase) => {
+  return fc
+    .integer()
+    .noBias()
```

Comment:
This function needs some comments.  Why `.noBias`?  Why `.noShrink`?  (Checking the documentation doesn't even give me a good idea what shrinking is.)  What is `faker`?  This all needs more explanation.